app/api/v2/resource/phones.rb

```diff
@@ -72,16 +77,21 @@ def validate_phone!(phone_number)
                    type: String,
                    allow_blank: false,
                    desc: 'Phone number with country code'
+          requires :channel,
+                   type: String,
+                   allow_blank: false,
+                   values: { value: %w(sms call), message: 'resource.phone.invalid_channel'},
+                   desc: 'The verification method to use'
         end
         post '/send_code' do
           declared_params = declared(params)
           validate_phone!(declared_params[:phone_number])
 
           phone_number = Phone.international(declared_params[:phone_number])
           phone = current_user.phones.find_by!(number: phone_number)
-          code_error!(phone.errors.details, 422) unless phone.regenerate_code
+          code_error!(phone.errors.details, 422) unless phone
```

Comment:
this doesn't make any sense now.
It was an error if the phone cant be updated. And now? This is invalid